### PR TITLE
boards: mimxrt10{50,60,64}_evk: Enable ft5336 touch interrupt

### PR DESCRIPTION
**Pull request by MaureenHelm**
Enables the ft5336 touch controller interrupt on the mimxrt10{50,60,64}_evk
boards.

Thanks to @k0d for adding interrupt support to the ft5336 driver in #23619.

### CODE REVIEW

**Review by jhqian — APPROVED** [review]

File: boards/arm/mimxrt1050_evk/pinmux.c [hunk 1]
```diff
@@ -109,6 +109,15 @@ static int mimxrt1050_evk_init(struct device *dev)
 	CLOCK_EnableClock(kCLOCK_Iomuxc);
 	CLOCK_EnableClock(kCLOCK_IomuxcSnvs);
 
+#if CONFIG_KSCAN_FT5336_INTERRUPT
```
Comment (galak) [inline boards/arm/mimxrt1050_evk/pinmux.c:112]: Can this be `#if DT_IRQ_HAS_IDX(DT_INST(0, kscan_ft5336), 0)`

File: boards/arm/mimxrt1060_evk/pinmux.c [hunk 2]
```diff
@@ -24,6 +24,15 @@ static int mimxrt1060_evk_init(struct device *dev)
 	CLOCK_EnableClock(kCLOCK_Iomuxc);
 	CLOCK_EnableClock(kCLOCK_IomuxcSnvs);
 
+#if CONFIG_KSCAN_FT5336_INTERRUPT
```
Comment (galak) [inline boards/arm/mimxrt1060_evk/pinmux.c:27]: see comment for mimxrt1050_evk pinmux.

**Review by alexandru-porosanu-nxp — APPROVED** [review]
LGTM

File: boards/arm/mimxrt1050_evk/pinmux.c (hunk 1 above)
Comment (MaureenHelm) [reply boards/arm/mimxrt1050_evk/pinmux.c]: This is a gpio, not an interrupts property. Updated to `#if DT_NODE_HAS_PROP(DT_INST(0, focaltech_ft5336), int_gpios)` instead.

**Pull request merged**